tests/test_mosaic.py

```diff
@@ -222,18 +221,64 @@ def test_mosaic_tiler_Stdev():
 
 def test_threads():
     """Test mosaic tiler."""
-    assets = [asset1, asset2, asset1, asset2, asset1, asset2]
+    assets = [asset2, asset1, asset1, asset2, asset1, asset2]
```

Comment:
Simple but important changes, before we were just lucky because the tiler was first yielding result for asset1. This change was made during testing to validate the new implementation